src/Microsoft.Health.Fhir.Api/Features/ActionResults/FhirResult.cs

```diff
@@ -78,5 +83,36 @@ public static FhirResult NoContent()
                 StatusCode = HttpStatusCode.NoContent,
             };
         }
+
+        /// <inheritdoc />
```

Comment:
One of the reason for the base class is to share this logic. Looks like `Result.ToPoco()` is the only difference between the two. Can we still share most of the code and just have the derived class provide that (could extend BaseActionResult to take additional type parameter?)